#!/usr/bin/env python2

PACKAGE = "Vision"
# The following line should not be here for Groovy and above
import roslib;roslib.load_manifest(PACKAGE)

# For Groovy and above, the following line should be
# from dynamic_reconfigure.parameter_generator_catkin import *
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("orange_hueLow",  int_t, 0, "Orange Hue Low ",    10, 0, 180)
gen.add("orange_hueHigh", int_t, 0, "Orange Hue High",   60, 0, 180)
gen.add("orange_satLow",  int_t, 0, "Orange Sat Low",  0, 0, 255)
gen.add("orange_satHigh", int_t, 0, "Orange Sat High", 100, 0, 255)
gen.add("orange_valLow",  int_t, 0, "Orange Val Low",    0, 0, 255)
gen.add("orange_valHigh", int_t, 0, "Orange Val High", 255, 0, 255)

gen.add("bin_area", int_t, 0, "Area to descend to", 100000, 0, 200000)
gen.add("shape_hu", double_t, 0, "Shape to identify", 0.17, 0, 50000)
gen.add("firing_x", double_t, 0, "Constant for forward move", 0.0005, 0, 1)
gen.add("firing_y", double_t, 0, "Constant for side move", 0.001, 0, 1)

gen.add("centering_x", double_t, 0, "Constant for forward move", 0.002, 0, 1)
gen.add("centering_y", double_t, 0, "Constant for side move", 0.004, 0, 1)

gen.add("aiming_x", double_t, 0, "Constant for forward move", 0.001, 0, 1)
gen.add("aiming_y", double_t, 0, "Constant for side move", 0.002, 0, 1)

exit(gen.generate(PACKAGE, "Drivethru", "Drivethru"))
